lib/kubernetes-deploy/runner.rb

```diff
@@ -76,11 +78,14 @@ def run
       phase_heading("Parsing deploy content")
       resources = discover_resources
 
+      tprs = resources.reject! { |r| r.tpr? }
+
       phase_heading("Checking initial resource statuses")
       resources.each(&:sync)
 
       phase_heading("Predeploying priority resources")
       predeploy_priority_resources(resources)
+      update_tprs(tprs)
```

Comment:
The single apply is intentional, but it is only essential when `prune: true`. Perhaps something like this? i.e. still have special treatment, but without circumventing the regular flow:

```ruby
def deploy_resources(resources, prune: false)
  tprs, resources = resources.partition(&:tpr?)
  deploy_tprs(tprs) if tprs.present?
 ...
```